haystack/nodes/file_converter/pdf.py

```diff
@@ -169,8 +169,8 @@ def _read_pdf(self, file_path: Path, layout: bool, encoding: Optional[str] = Non
         if not encoding:
             encoding = self.encoding
 
-        command = f"pdftotext -enc {encoding} {'-layout ' if layout else ''}{str(file_path)} -".split()
-        output = subprocess.run(command, stdout=subprocess.PIPE, shell=False)
+        command = ["pdftotext", "-enc", str(encoding), "-layout" if layout else "-raw", str(file_path), "-"]
+        output = subprocess.run(command, stdout=subprocess.PIPE, shell=False, check=False)
```

Comment:
Why `check=False`? According to [the docs for `subprocess.run`](https://docs.python.org/3/library/subprocess.html#subprocess.run):

> If _check_ is true, and the process exits with a non-zero exit code, a `CalledProcessError` exception will be raised.

Sounds like a good thing to do. Or does it cause problems?

Reply:
I tried to not modify code behavior. Default value for `check` is already _False_, but it's recommended that user explicitly set it (an orientation on newer versions).

If check is set to True, there will need to be a try catch block. Shoud I implemented ? It would be better to log an error like this.